### PR TITLE
Allow code generation for multiple API versions

### PR DESCRIPTION
**Pull request by tnozicka**
**Description of your changes:**
This PR allows to generate deepcopy, clients, listers, informers and crds for multiple API versions. Needed by the future PRs that introduce new v1alpha1 types.

PoC in https://github.com/scylladb/scylla-operator/pull/782

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -232,20 +247,42 @@ update-codegen:
...
+
 endef
 
-update-crd:
-	$(call generate-crd,$(CRD_PATH))
-.PHONY: update-crd
+# $1 - dir prefix
+define generate-crds
+	$(foreach p,$(api_package_dirs),$(call run-crd-gen,$(p),$(1)$(p)))
+	find $(foreach p,$(api_package_dirs),$(1)$(p)) -mindepth 1 -maxdepth 1 -type f -name '*.yaml' -exec $(YQ) -i eval '.' {} \;
```
Comment (zimnx) [inline Makefile:268]: shouldn't this be part of verify-crds?

Comment (zimnx) [reply Makefile:268]: oh, this is needed to reformat output files in yq formatting style right?

File: Makefile [hunk 2]
```diff
@@ -44,7 +43,19 @@ GO_TEST_ARGS ?=
...
 GO_TEST_E2E_EXTRA_ARGS ?=
 
-YQ ?=yq
+YQ ?=yq -e
+
+CODEGEN_PKG ?=./vendor/k8s.io/code-generator
+CODEGEN_HEADER_FILE ?=/dev/null
+
+# API_PACKAGES is a list of comma separated full api package refs (like golang import strings)
+API_PACKAGES ?=$(shell $(GO) list -json ./pkg/api/scylla/... | jq -sr 'map(select(.Name | test("^v[0-9]+.*$$"))) | reduce .[] as $$item ([]; . + [$$item.ImportPath]) | join(",")')$(if $(filter $(.SHELLSTATUS),0),,$(error failed to list api packages))
```
Comment (zimnx) [inline Makefile:52]: `jq` -> `$(JQ)`

File: Makefile (hunk 1 above)
Comment (tnozicka) [reply Makefile:268]: yep, it looses the `---`

Comment (tnozicka) [reply Makefile:268]: yep, but it looks like the comment got left in the other macro

Comment (tnozicka) [reply Makefile]: I've moved the find to the comment

File: Makefile (hunk 2 above)
Comment (tnozicka) [reply Makefile]: done, thx

**Review by zimnx — APPROVED** [review]

**Pull request merged**